### ISSUE TITLE
Extremely long validation timeouts making DBeaver almost unusable

### ISSUE THREAD

**Issue by ssadedin** [dbeaver/dbeaver#4807]
I'm having trouble using DBeaver because the following scenario happens:

 - I connect to the database and run a query
 - I go away and do something else
 - now the database connection / network connection (over VPN) has timed out
 - I come back to run a query
 - DBeaver hangs and is unresponsive for up to 2 minutes before it invalidates the connection

I have set the connection validation timeout to 3 seconds, but it still takes minutes to recover the connection. During that time I can't establish a new connection or do anything useful with DBeaver.

Is there a way to have it just completely drop a connection after a few minutes of idle time instead of trying to keep it alive? I think my networking setup probably plays a role in all this such that the database connection might look alive and stops DBeaver recognising that it needs to invalidate for much longer than it should.

How can I get better behavior?

**Comment by serge-rider** [comment]
What version of DBeaver do you use?
We redesigned connection invalidate function in one of recent versions.

**Comment by uslss** [comment]
hi @ssadedin ,
is issue still actual for you?

**Comment by ssadedin** [comment]
I haven't actually updated my version yet - I actually get this error when I check for updates:

![image](https://user-images.githubusercontent.com/138868/66243392-cd9fb180-e747-11e9-80f4-acdddc316ed2.png)

The problem is the same of course in the version I'm using. The sceanrio is:

- I open DBeaver and run a query
- I go back to my coding for a bit, 20 mins later I want to try something again in DBeaver
- I execute a query, but now the connection is hung, so what I get is a huge long hang that takes minutes to expire, and in that time I cannot use DBeaver at all. 

Here is one I did just now after doing a query 1 hour ago - it is still hung after 3 minutes and I cannot interact with that database at all with DBeaver because of the hung connection:

![image](https://user-images.githubusercontent.com/138868/66243451-0cce0280-e748-11e9-8d96-25ff9eb2828c.png)

**Comment by ssadedin** [comment]
NB: so after something like 4 minute it finally timed out, then I see this:

![image](https://user-images.githubusercontent.com/138868/66243577-5fa7ba00-e748-11e9-8bfd-c998b3ae8dbb.png)

Then the query executes instantly.

**Comment by serge-rider** [comment]
@ssadedin Try to upgrade to a more recent version. 5.2 is way too outdated.
We added invalidate/reconnect timeouts (configurable, 10 sec by default) in some old version so now it shouldn't take so much time.

**Comment by ssadedin** [comment]
Hi @serge-rider I gave 6.2 a shot - still same behavior I think.

I started a session yesterday, left it 24 hours, during which my whole VPN connection to the database was dropped. Then I went into the query window and tried to execute a query and it hung for almost a minute:

![image](https://user-images.githubusercontent.com/138868/66264240-52c0be80-e84d-11e9-91cc-31e764a7471c.png)

Then eventually it popped up the "Datasource was invalidated" message above ... but only after locking me out of using the tool for all that time. This is happening even though I have these settings in preferences:

![image](https://user-images.githubusercontent.com/138868/66264266-b2b76500-e84d-11e9-8f56-8a3f5062ce0c.png)

So I don't understand why, with those timeout settings, it is making me wait so long before it invalidates the connection.
